rdbms/src/main/java/org/spine3/server/storage/jdbc/JdbcAggregateStorage.java

```diff
@@ -0,0 +1,228 @@
+ * @see JdbcStorageFactory
+ * @author Alexander Litus
+ */
+/*package*/ class JdbcAggregateStorage<ID> extends AggregateStorage<ID> {
+
+    @SuppressWarnings({"UtilityClass", "DuplicateStringLiteralInspection", "ClassNamingConvention"})
+    private static class SQL {
+
+        /**
+         * Aggregate ID column name.
+         */
+        private static final String ID = "id";
+
+        /**
+         * Aggregate record column name.
+         */
+        private static final String AGGREGATE = "aggregate";
+
+        /**
+         * Aggregate event seconds column name.
+         */
+        private static final String SECONDS = "seconds";
+
+        /**
+         * Aggregate event nanoseconds column name.
+         */
+        private static final String NANOSECONDS = "nanoseconds";
+
+        private static final String INSERT =
+                "INSERT INTO %s " +
+                " (" + ID + ", " + AGGREGATE + ", " + SECONDS + ", " + NANOSECONDS + ") " +
+                " VALUES (?, ?, ?, ?);";
+
+        private static final String SELECT_BY_ID_SORTED_BY_TIME_DESC =
+                "SELECT " + AGGREGATE + " FROM %s " +
+                " WHERE " + ID + " = ? " +
+                " ORDER BY " + SECONDS + " DESC, " + NANOSECONDS + " DESC;";
+
+        private static final String CREATE_TABLE_IF_DOES_NOT_EXIST =
+                "CREATE TABLE IF NOT EXISTS %s (" +
+                    ID + " %s, " +
+                    AGGREGATE + " BLOB, " +
+                    SECONDS + " BIGINT, " +
+                    NANOSECONDS + " INT " +
+                ");";
+    }
+
+    private static final Descriptor RECORD_DESCRIPTOR = AggregateStorageRecord.getDescriptor();
+
+    private final DataSourceWrapper dataSource;
+
+    /**
+     * Iterators which are not closed yet.
+     */
+    private final Collection<DbIterator> iterators = newLinkedList();
+
+    private final IdHelper<ID> idHelper;
+
+    /**
+     * SQL queries.
+     */
+    private final String insertQuery;
+    private final String selectByIdSortedByTimeDescQuery;
+
+    /**
+     * Creates a new storage instance.
+     *
+     * @param dataSource the dataSource wrapper
+     * @param aggregateClass the class of aggregates to save to the storage
+     */
+    /*package*/ static <ID> JdbcAggregateStorage<ID> newInstance(DataSourceWrapper dataSource,
+                                                     Class<? extends Aggregate<ID, ?>> aggregateClass) {
+        return new JdbcAggregateStorage<>(dataSource, aggregateClass);
+    }
+
+    private JdbcAggregateStorage(DataSourceWrapper dataSource, Class<? extends Aggregate<ID, ?>> aggregateClass) {
+        this.dataSource = dataSource;
+
+        final String tableName = DbTableNameFactory.newTableName(aggregateClass);
+        this.insertQuery = format(SQL.INSERT, tableName);
+        this.selectByIdSortedByTimeDescQuery = format(SQL.SELECT_BY_ID_SORTED_BY_TIME_DESC, tableName);
```

Comment:
If you wrap constants from `SQL` together with formatting logic and execution of those queries, you'd improve encapsulation.